### PR TITLE
Implement simpler hash function

### LINKED ISSUE #7: Implement simpler hash function
The current hashing routine to verify script bytecode is overly complex. Simplify by implementing in assembly instead

### PR DESCRIPTION
**Pull request by pixelcircuits**
Moved helper functions from the predicate into their own utils library and added a new helper function `tx_script_bytecode_hash` to easily get an unpadded script hash. 

Note: there is a future issue to tweak this new function to use GTF in the future

### CODE REVIEW

**Review by Braqzen — COMMENTED** [review]
1. `main.sw`
    - Has unused imports
        - `ContractId`
        - `INPUT_MESSAGE`
    - `MIN_GAS` const can probably be moved into the manifest file. The latest `forc` allows us to load values from the file as a pseudo constructor. I haven't used this myself yet
2. `utils.sw`
    - Has unused imports
        - `OUTPUT_CHANGE`
        - `OUTPUT_CONTRACT`
        - `OUTPUT_VARIABLE`
        - `tx_gas_limit`
        - `tx_inputs_count`
        - `tx_output_type`
        - `tx_outputs_count`
        - `sha256`
    - Similar comment regarding moving constants into manifest
    - I would keep all the imports together and then below them declare the constants (if not put in manifest) rather than have some imports at the top and then a `core::num` and constants somewhere at the bottom of the file. Keeping things neatly together, at the top, makes it easier to navigate when searching

**Comment by adlerjohn** [comment]
If there isn't an issue in the Sway repo to warn on unused imports then please file one.

File: contract-message-predicate/src/main.sw [hunk 1]
```diff
@@ -59,17 +32,14 @@ fn main() -> bool {
 
     // The hash of the script which must spend the input belonging to this predicate
     // This ensures the coins can only be spent in a call to `TokenContract.finalizeDeposit()`
-    // Note: The script must be right-padded to the next full word before hashing, to match with `get_script_bytecode()`
-    const SPENDING_SCRIPT_HASH = 0x2d235589506d17993e0b7aca4407a5ac1c325efd9d704ff94696a8f7c012ab9d;
+    const SPENDING_SCRIPT_HASH = 0x6856bb03b84d876ee60c890be5b0602d0f7480d375917a660da3115e8e008ddb;
```
Comment (adlerjohn) [inline contract-message-predicate/src/main.sw:35]: This seems like the perfect usecase for configuration-time constants!

Comment (adlerjohn) [reply contract-message-predicate/src/main.sw:35]: Could be deferred to a subsequent PR.

**Review by adlerjohn — COMMENTED** [review]
New versions of `forc` and `fuels-rs` are around the corner. Would it be better to wait a few days and do this properly with the new functions instead of hard-coding things that will have to get changed later?

**Comment by pixelcircuits** [comment]
> If there isn't an issue in the Sway repo to warn on unused imports then please file one.

Someone already made an issue for it: https://github.com/FuelLabs/sway/issues/1298

**Comment by pixelcircuits** [comment]
> New versions of `forc` and `fuels-rs` are around the corner. Would it be better to wait a few days and do this properly with the new functions instead of hard-coding things that will have to get changed later?

Yeah, I think I'm gonna scrap this PR since I've started working with using the new GTF refactor. Merging this minor issue may not be worth the effort

**Comment by pixelcircuits** [comment]
Closing to prioritize new sway and sdk releases

**Pull request closed without merging**